pkg/networkpolicy/networkpolicy.go

```diff
@@ -10,6 +10,29 @@ import (
 	"k8s.io/klog/v2"
 )
 
+// getPodsForNetworkPolicyNode obtains all the Pods selected by the network policy for current Node
+func (c *Controller) getLocalPodsForNetworkPolicy(networkPolicy *networkingv1.NetworkPolicy) []*v1.Pod {
+	if networkPolicy == nil {
+		return nil
+	}
+	podSelector, err := metav1.LabelSelectorAsSelector(&networkPolicy.Spec.PodSelector)
```

Comment:
So FTR, with the existing code, if you have Pod A on Node 1 that denies all ingress, and Pod B on Node 2 tries to send to it, then Node 2 will drop the packets. With this new code, Node 2 would instead forward the packet to Node 1, and then Node 1 would drop it.

Reply:
yeah, is the trade off, but at least the rules are only evaluated once in one node, previously allowed rules were evaluated twice